pandas/core/construction.py

```diff
@@ -379,6 +381,16 @@ def array(
     ):
         return TimedeltaArray._from_sequence(data, dtype=dtype, copy=copy)
 
+    elif lib.is_np_dtype(dtype, "mM"):
+        warnings.warn(
+            r"Deprecating unsupported datetime64 and timedelta64 dtype resolutions. "
```

Comment:
the first sentence here is awkward.  Maybe "datetime64 and timedelta64 dtypes with resolutions other than ... are deprecated."

If they pass an unsupported dtype, we should raise, not silently cast